### PR TITLE
Add canonicalize option to MolecularFeaturizer

### PR DESCRIPTION
**Pull request by nd-02110114**
I added canonicalize option to MolecularFeaturizer and update docs.
The diff is big, but the reason is just reverting this commit. https://github.com/deepchem/deepchem/commit/f7c12790fe0d9301e4df5b49ef8bf97019f7b51e I made a mistake about merging this commit in #2088.

### CODE REVIEW

File: deepchem/feat/base_classes.py [hunk 1]
```diff
@@ -159,59 +147,60 @@ class MolecularFeaturizer(Featurizer):
...
-          # so that they're always in the same canonical order.
-          # Presumably this should be correctly implemented in the
-          # future for graph mols.
-          if mol:
-            new_order = rdmolfiles.CanonicalRankAtoms(mol)
-            mol = rdmolops.RenumberAtoms(mol, new_order)
+        # canonicalize
+        if canonical:
+          canonical_smiles = Chem.MolToSmiles(mol)
+          mol = Chem.MolFromSmiles(canonical_smiles)
```
Comment (nd-02110114) [inline deepchem/feat/base_classes.py:202]: I add canonicalize option to `MolecularFeaturizer`

Comment (nd-02110114) [reply deepchem/feat/base_classes.py:202]: Yes. Please confirm this documents.
https://www.rdkit.org/docs/GettingStartedInPython.html#writing-molecules

I think the official canonicalize approach is like this.

```
>>> Chem.MolToSmiles(Chem.MolFromSmiles('C1=CC=CN=C1'))
'c1ccncc1'
>>> Chem.MolToSmiles(Chem.MolFromSmiles('c1cccnc1'))
'c1ccncc1'
>>> Chem.MolToSmiles(Chem.MolFromSmiles('n1ccccc1'))
'c1ccncc1'
```

Comment (rbharath) [reply deepchem/feat/base_classes.py:202]: Ok looks good! Feel free to merge in once ready

Comment (nd-02110114) [reply deepchem/feat/base_classes.py]: I noticed this implementation loses coordinate information if mol object has them ...
So, I went back to original implementation.

**Comment by nd-02110114** [comment]
Currently, this PR is just modifying the docstring, so I merge.

**Pull request merged**